Begin Preparing Model Card Guidebook content. (#545)
huggingface/hub-docs: docs/hub/model-card-annotated.md

* The **sociotechnic** is necessary for filling out “Bias” and “Risks” within [Bias, Risks, and Limitations](#bias-risks-and-limitations), and particularly useful for “Out of Scope Use” within [Uses](#uses).

* The **project organizer** is necessary for filling out [Model Details](#model-details) and [Uses](#uses). They might also fill out
[Training Data](#training-data). Project organizers could also be in charge of [Citation](#citation), [Glossary](#glossary), 
[Model Card Contact](#model-card-contact), [Model Card Authors](#model-card-authors), and [More Information](#more-information).

_Instructions are provided below, in italics._

# Model Name

**Section Overview:**  Provide the model name and a 1-2 sentence summary of what the model is. 

{{ model_id | default("Model ID", true) }}
{{ model_summary | default("", true) }}

# Table of Contents

> **Section Overview:** Provide this with links to each section, to enable people to easily jump around/use the file in other locations with the preserved TOC/print out the content/etc.

Reviewer: ```suggestion

**Section Overview:** Provide this with links to each section, to enable people to easily jump around/use the file in other locations with the preserved TOC/print out the content/etc.
```

jump a line otherwise it's not rendered properly (cf. https://moon-ci-docs.huggingface.co/docs/hub/pr_545/en/model-card-annotated)

# Model Details

**Section Overview:** This section provides basic information about what the model is, its current status, and where it came from. It should be useful for anyone who wants to reference the model.

## Model Description

> {{ model_description | default("", true) }}

Reviewer: we should remove the curly braces syntax and just leave the textual description for each section, no?

Author: I think it's helpful to have them to help people just visually align between what they see in the usual template and what the annotations are. It also tells people the variable name of what the instructions are referring to, which seems useful.

_Name, and link to, the license being used._

* **Finetuned From Model [optional]:** {{ finetuned_from | default("[More Information Needed]", true)}}

_If this model has another model as its base, link to that model here._

## Model Sources [optional]

* **Repository:** {{ repo | default("[More Information Needed]", true)}}
* **Paper [optional]:** {{ paper | default("[More Information Needed]", true)}}
* **Demo [optional]:** {{ demo | default("[More Information Needed]", true)}}

_Provide sources for the user to directly see the model and its details. Additional kinds of resources – training logs, lessons learned, etc. – belong in the [More Information](#more-information) section. If you include one thing for this section, link to the repository._

# Uses
**Section Overview:** This section addresses questions around how the model is intended to be used, discusses the foreseeable users of the model (including those affected by the model), and describes uses that are considered out of scope or misuse of the model.  Note this section is not intended to include the license usage details. For that, link directly to the license.

> ## Direct Use

Reviewer 2: The difference between Direct Use and Downstream use are not very clear to me. 

On one, we would want to explain how to use the model for fine-tuning.
On the other, we would like to show how to use the model without fine-tuning (e.g. for running inference in an app).

Is that right? The descriptions seem to mix a bit both.

Author: Hmm. One says "without fine-tuning", then the other says "when fine-tuned". Do you have a wording suggestion to make it less confusing?

Author: (Explaining how to fine-tune the model isn't what's happening here: this section is for how the model is used in different applied contexts.)

Author: Added "in different applied contexts". Does that help at all?

Reviewer 2: Yep this is good. I think this is good!

{{ direct_use | default("[More Information Needed]", true)}}

> _List how the model can be used without fine-tuning, post-processing, or plugging into a pipeline._

Reviewer 2: ```suggestion
_Explain how the model can be used without fine-tuning, post-processing, or plugging into a pipeline. An example code snippet is recommended._
```

## Downstream Use [optional]

{{ downstream_use | default("[More Information Needed]", true)}}

> _List how this model can be used when fine-tuned for a task, or when plugged into a larger ecosystem or app._

Reviewer 2: ```suggestion
_Explain how this model can be used when fine-tuned for a task or when plugged into a larger ecosystem or app. An example code snippet is recommended._
```

## Out-of-Scope Use

{{ out_of_scope_use | default("[More Information Needed]", true)}}

_List how the model may foreseeably be misused and address what users ought not do with the model._

# Bias, Risks, and Limitations

> **Section Overview:** This section identifies foreseeable harms and misunderstandings, and both technical and sociotechnical limitations. It also provides information on warnings and potential mitigations.

Reviewer 2: ```suggestion
**Section Overview:** This section identifies foreseeable harms, misunderstandings, and technical and sociotechnical limitations. It also provides information on warnings and potential mitigations.
```

 

## Training Procedure [optional]

### Preprocessing

{{ preprocessing | default("[More Information Needed]", true)}}

### Speeds, Sizes, Times

{{ speeds_sizes_times | default("[More Information Needed]", true)}}

_Detail throughput, start/end time, checkpoint sizes, etc._

# Evaluation

**Section Overview:** This section describes the evaluation protocols, what is being measured in the evaluation, and provides the results.  Evaluation is ideally constructed with factors, such as domain and demographic subgroup, and metrics, such as accuracy, which are prioritized in light of foreseeable error contexts and groups. Target fairness metrics should be decided based on which errors are more likely to be problematic in light of the model use. 

## Testing Data, Factors & Metrics

### Testing Data

{{ testing_data | default("[More Information Needed]", true)}}

_Ideally this links to a Dataset Card for the testing data._

### Factors

{{ testing_factors | default("[More Information Needed]", true)}}

> _What are the foreseeable characteristics that will influence how the model behaves? This includes domain and context, as well as population subgroups. Evaluation should ideally be **disaggregated** across factors._

Reviewer 2: Maybe giving an example on this would be useful

Author: Added as a suggestion!

Author: Incorporated!

# Annotated Model Card Template

## Template

[https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/modelcard_template.md](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/modelcard_template.md)

## Directions

Fully filling out a model card requires input from a few different roles. (One person may have more than one role.)  We’ll refer to these roles as the **developer**, who writes the code and runs training; the **sociotechnic**, who is skilled at analyzing the interaction of technology and society long-term (this includes lawyers, ethicists, sociologists, or rights advocates); and the **project organizer**, who understands the overall scope and reach of the model, can roughly fill out each part of the card, and who serves as a contact person for model card updates.

> * The **developer** is necessary for filling out [Training Procedure](#training-procedure) and [Technical Specifications](#technical-specifications). They are also particularly useful for the “Limitations” section of [Bias, Risks, and Limitations](#bias-risks-andlimitations). They are responsible for providing [Results](#results) for the Evaluation, and ideally work with the other roles to define the rest of the Evaluation: [Testing Data, Factors & Metrics](#testing-data-factors-and-metrics).

Reviewer 2: I found links to other sections broken in multiple sections. For the right link, you can go to https://moon-ci-docs.huggingface.co/docs/hub/pr_545/en/model-card-annotated#testing-data-factors-metrics and hover over the title. There will be a copy link option, which will help fix these

```suggestion
* The **developer** is necessary for filling out [Training Procedure](#training-procedure-optional) and [Technical Specifications](#technical-specifications-optional). They are also particularly useful for the “Limitations” section of [Bias, Risks, and Limitations](#bias-risks-and-limitations). They are responsible for providing [Results](#results) for the Evaluation, and ideally work with the other roles to define the rest of the Evaluation: [Testing Data, Factors & Metrics](#testing-data-factors-metrics).
```

Author: Oh, thank you for this! I hadn't gone back through those links to check after first just jotting placeholders. 🙌

* The **sociotechnic** is necessary for filling out “Bias” and “Risks” within [Bias, Risks, and Limitations](#bias-risks-and-limitations), and particularly useful for “Out of Scope Use” within [Uses](#uses).

* The **project organizer** is necessary for filling out [Model Details](#model-details) and [Uses](#uses). They might also fill out
[Training Data](#training-data). Project organizers could also be in charge of [Citation](#citation), [Glossary](#glossary), 
[Model Card Contact](#model-card-contact), [Model Card Authors](#model-card-authors), and [More Information](#more-information).

_Instructions are provided below, in italics._

Template variable names appear in `monospace`.

# Model Name

**Section Overview:**  Provide the model name and a 1-2 sentence summary of what the model is. 

`model_id`

`model_summary`

> # Table of Contents

Reviewer 2: Just a friendly reminder that we automatically create ToCs on the Hub and that we can modify the print mechanism so it's generated automatically on print. Keeping it here is fine though I guess

Author: I think that's useful to point out, although I wonder the best way to point it out without making it too cluttered.
In the interest of getting this out, I'm going to not resolve this conversation + we can return to it later if we're feeling up for it.

**Section Overview:** Provide this with links to each section, to enable people to easily jump around/use the file in other locations with the preserved TOC/print out the content/etc.

# Model Details

**Section Overview:** This section provides basic information about what the model is, its current status, and where it came from. It should be useful for anyone who wants to reference the model.

## Model Description

`model_description`

_Provide basic details about the model. This includes the architecture, version, if it was introduced in a paper, if an original implementation is available, and the creators. Any copyright should be attributed here. General information about training procedures, parameters, and important disclaimers can also be mentioned in this section._

* **Developed by:** `developers`

_List (and ideally link to) the people who built the model._

* **Shared by [optional]:** `shared_by`

_List (and ideally link to) the people/organization making the model available online._

* **Model type:** `model_type`

_You can name the “type” as:_

_1. Supervision/Learning Method_

_2. Machine Learning Type_

_3. Modality_

* **Language(s)** [NLP]: `language`

> _Use this field when the system is a natural language processing system._

Reviewer 2: Could potentially be for `text-to-image` and ML for audio (e.g. text to speech) too

Author: That's a great point; NLP here is being used more as a descriptor term rather than a field term, but that's indistinguishable! I can change to `...when the system uses or processes natural (human) language`.

* **License:** `license`

> _Name, and link to, the license being used._

Reviewer 2: ```suggestion
_Name and link to the license being used._
```

## Training Data

`training_data`

_Write 1-2 sentences on what the training data is. Ideally this links to a Dataset Card for further information. Links to documentation related to data pre-processing or additional filtering may go here as well as in [More Information](#more-information)._
 

## Training Procedure [optional]

### Preprocessing

`preprocessing`

_Detail tokenization, resizing/rewriting (depending on the modality), etc._

### Speeds, Sizes, Times

`speeds_sizes_times`

_Detail throughput, start/end time, checkpoint sizes, etc._

# Evaluation

**Section Overview:** This section describes the evaluation protocols, what is being measured in the evaluation, and provides the results.  Evaluation is ideally constructed with factors, such as domain and demographic subgroup, and metrics, such as accuracy, which are prioritized in light of foreseeable error contexts and groups. Target fairness metrics should be decided based on which errors are more likely to be problematic in light of the model use. 

## Testing Data, Factors & Metrics

### Testing Data

`testing_data`

_Ideally this links to a Dataset Card for the testing data._

### Factors

`testing_factors`

> _What are the foreseeable characteristics that will influence how the model behaves? This includes domain and context, as well as population subgroups. Evaluation should ideally be **disaggregated** across factors._

Author: ```suggestion
_What are the foreseeable characteristics that will influence how the model behaves? This includes domain and context, as well as population subgroups. Evaluation should ideally be **disaggregated** across factors in order to uncover disparities in performance._
```
